File: FSUIKit/Classes/FSLabel.m
//
//  FSLabel.m
//  Expand
//
//  Created by Fudongdong on 2017/8/4.
//  Copyright © 2017年 china. All rights reserved.
//

#import "FSLabel.h"

@implementation FSLabel

- (instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        self.numberOfLines = 0;
    }
    return self;
}

- (void)setText:(NSString *)text{
    [super setText:text];
    
    CGRect frame = self.frame;
    [self sizeToFit];
    self.frame = CGRectMake(frame.origin.x, frame.origin.y, frame.size.width, self.frame.size.height);
}

- (void)setFont:(UIFont *)font{
    [super setFont:font];
    [self setText:self.text];
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: FSUIKit/Classes/Swift/FSLabel.swift
// FSLabelS.swift
// Translated from FSLabel.h/m

import UIKit

public class FSLabel: UILabel {
    
    public override init(frame: CGRect) {
        super.init(frame: frame)
        numberOfLines = 0
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        numberOfLines = 0
    }
    
    public override var text: String? {
        didSet {
            let savedFrame = frame
            sizeToFit()
            frame = CGRect(x: savedFrame.origin.x, y: savedFrame.origin.y, width: savedFrame.width, height: frame.height)
        }
    }
    
    public override var font: UIFont! {
        didSet {
            // Trigger text setter to recalculate size
            let currentText = text
            text = currentText
        }
    }
}
